public/scripts/controllers/AlbumsIndexController.js

```diff
@@ -0,0 +1,67 @@
+angular
+  .module('tunely')
+  .controller('AlbumsIndexController', AlbumsIndexController);
+
+AlbumsIndexController.$inject = ['$http'];
+
+function AlbumsIndexController ($http) {
+  var vm = this;
+  vm.newAlbum = {};
+  vm.newAlbum = {
+    name: 'Viva Hate',
+    artistName: 'Morrissey'
+  };
+
+  $http({
+    method: 'GET',
+    url: '/api/albums'
+  }).then(function successCallback(response) {
+    vm.albums = response.data;
+  }, function errorCallback(response) {
+    console.log('There was an error getting the data', response);
+  });
+
+  vm.createAlbum = function () {
+    $http({
+      method: 'POST',
+      url: '/api/albums',
+      data: vm.newAlbum,
+    }).then(function successCallback(response) {
+      vm.albums.push(response.data);
+    }, function errorCallback(response) {
+      console.log('There was an error posting the data', response);
+    });
+  }
+
+  vm.editAlbum = function (album) {
```

Comment:
See here for a similar approach: https://github.com/sf-wdi-25/angular_book_app/blob/solution/controllers.js By the end of coding this with students we started discussing SoC and it was clear to us that a refactor was in order (due to overloading the index controller).